source/projectile/rocket/Rocket.hx

```diff
@@ -0,0 +1,43 @@
+package projectile.rocket;
+
+import flixel.group.FlxGroup;
+import flixel.math.FlxPoint;
+
+class Rocket extends Projectile {
+	private static var SPEED = -25;
+	private static var ACCEL = 1500;
+	private static var RESIZE:Float = 2.5;
+
+	private static var ROCKET_GRAPHIC:String = "assets/images/rocket.png";
+
+	public var explosion:Explosion;
+
+	override public function new() {
+		explosion = new Explosion();
+		super();
+		loadGraphic(ROCKET_GRAPHIC);
+		setSize(RESIZE, RESIZE);
+	}
+
+	override public function update(elapsed:Float) {
+		super.update(elapsed);
+	}
+
+	override public function fire(startPoint:FlxPoint, aimDegrees:Float) {
+		super.fire(startPoint, aimDegrees);
+		var baseAccel = FlxPoint.weak(1, 0);
+		acceleration = baseAccel.rotateByDegrees(aimDegrees).scale(ACCEL);
+	}
+
+	override public function impact(surface:String) {
+		explosion.fire(this.getMidpoint(), 0);
+		kill();
+	}
+
+	public function getObjectBundle() {
+		var group = new FlxGroup();
+		group.add(this);
+		group.add(explosion);
+		return group;
+	}
```

Comment:
This is similar to what we did for tanks, but I think it ends up being difficult to use as-is. 

Projectile and SimpleBullet don't have this `getObjectBundle()` method, so we have to do special logic and code to make something use these rockets compared to using `SimpleBullet`s.

The idea that missiles can fire explosions is a neat way to make the whole system work (and it means that adding the whole `getObjectBundle()` group to a group of bullets should still function properly)! 

Making the way we use the bullets consistent is important, though, without that our project becomes much harder to manage. I'm down to talk through this whole thing over voice chat if you'd like, I think it's a very valuable learning opportunity.

Reply:
We'll put this on hold for now, can revisit it after we've shown how to integrate bullets with the rest of the game